testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestAvroSchemaEvolution.java

```diff
@@ -0,0 +1,283 @@
+                        createRow(0, "stringA0", null));
+    }
+
+    @Test(groups = AVRO)
+    public void testSchemaWhenUrlIsUnset()
+    {
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("string_col", "varchar", "", ""),
+                                row("int_col", "integer", "", "")));
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        createRow(0, "stringA0", 0));
+
+        onHive().executeQuery(format("ALTER TABLE %s UNSET TBLPROPERTIES('avro.schema.url')", tableName));
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(
+                                row("dummy_col", "varchar", "", "")));
+    }
+
+    protected void alterTableSchemaTo(String schema)
+    {
+        onHive().executeQuery(format("ALTER TABLE %s SET TBLPROPERTIES('avro.schema.url'='%s')", tableName, schema));
+    }
+
+    private void insertData(String tableName, int rowNo, String... sqlValues)
+    {
+        String columnValues = Joiner.on(", ").join(sqlValues) +
+                (partitions.isEmpty() ? "" : ", '" + getPartitionsAsString("', '", p -> p.getValueForRow(rowNo)) + "'");
+        query(format("INSERT INTO %s VALUES (%s)", tableName, columnValues));
+    }
+
+    private Row createRow(int rowNumber, Object... data)
+    {
+        List<Object> rowData = new ArrayList<>(Arrays.asList(data));
+        partitions.forEach(partition -> rowData.add(partition.getValueForRow(rowNumber)));
+        return new Row(rowData);
+    }
+
+    private List<Row> prepareShowColumnsResultRows(Row... rows)
+    {
+        List<Row> rowsWithPartitions = new ArrayList<>(Arrays.asList(rows));
+        rowsWithPartitions.addAll(partitions.stream()
+                .map(partition -> row(partition.getName(), partition.getType(), "partition key", ""))
+                .collect(Collectors.toList()));
+        return rowsWithPartitions;
+    }
+
+    private String getPartitionsAsString(String joiningDelimiter, Function<Partition, String> partitionMapper)
+    {
+        return partitions.stream()
+                .map(partitionMapper)
+                .collect(Collectors.joining(joiningDelimiter));
+    }
+
+    protected static class Partition
+    {
+        private String name;
+        private String type;
+
+        public Partition(String name, String type)
+        {
+            this.name = requireNonNull(name, "name is null");
+            this.type = requireNonNull(type, "type is null")
+                    .toLowerCase(ENGLISH);
+        }
+
+        public String getName()
+        {
+            return name;
+        }
+
+        public String getType()
+        {
+            return type;
+        }
+
+        public String getColumnDeclaration()
```

Comment:
Similarly can we have a `String getDelimitedName` -> where we could wrap the column name like `'name'` and `getPartitionsAsString` can be simplified with 

```
    private String getPartitionsAsString(Function<Partition, String> partitionMapper)
    {
        return partitions.stream()
                .map(partitionMapper)
                .collect(joining(", "));
    }
```

Reply:
Done + renamed to getPartitionsAsListString